### ISSUE TITLE
Feature request: Parlleal the reference model compute and reward_fn calling

### ISSUE THREAD

**Issue by Question406** [volcengine/verl#324]
Currently, the computation of reference/critic model logits and reward_fn calling is in sequential order, but they can in theory in parallel:

https://github.com/volcengine/verl/blob/9448762515b73df24183e2bd6d24bb55d4989c55/verl/trainer/ppo/ray_trainer.py#L887-L914

Something in my mind is to do async call for the reward_fn, like the following:

```python
tasks = [
    asyncio.create_task(
        asyncio.to_thread(
            self.actor_rollout_wg.compute_log_prob, batch
        )
    ),
    asyncio.create_task(self.reward_fn(batch)),
]
if self.use_reference_policy:
    tasks.append(
        asyncio.create_task(
            asyncio.to_thread(
                self.ref_policy_wg.compute_ref_log_prob, batch
            )
        )
    )
if self.use_critic:
    tasks.append(
        asyncio.create_task(
            asyncio.to_thread(
                self.critic_wg.compute_values, batch
            )
        )
    )
results = await asyncio.gather(*tasks)
```

I can open a pull request if it is a doable feature.

**Comment by Kirrito-k423** [comment]
中文：感谢这个建议。当前 `verl/trainer/ppo/ray_trainer.py` 中 reward 提取、old log prob、reference log prob 和 critic values 仍然是在训练 step 内按顺序执行的，所以这个优化方向看起来仍然是相关的。

如果提交 PR，建议把实现做成保守且可配置的改动，并特别验证这些点：并行任务之间不要共享会被原地修改的 `DataProto`/batch 对象；异常、超时和取消语义要清晰；`use_reference_policy`、`use_critic`、`use_rm`、colocate reward model、以及不同 rollout/recompute 配置下的指标和结果应保持一致；最好附上性能对比和回归测试。

English: Thanks for the suggestion. In the current `verl/trainer/ppo/ray_trainer.py`, reward extraction, old log prob recomputation, reference log prob computation, and critic value computation still appear to run sequentially inside the training step, so this optimization direction still looks relevant.

If you open a PR, please keep the change conservative and preferably configurable. The PR should carefully validate that parallel tasks do not share mutable `DataProto`/batch state unsafely, that exception/timeout/cancellation behavior is well defined, and that metrics/results remain consistent across `use_reference_policy`, `use_critic`, `use_rm`, colocated reward model, and different rollout/recompute settings. A small benchmark plus regression coverage would make the proposal easier to review.

This response was generated by a scheduled Codex GPT5.5 issue processing task.
